### PR TITLE
Update the softmax default to axis=-1

### PR DESCRIPTION
**Pull request by mattdangerw**
Essentially we are going to adopt the jax approach for softmax:

- The `axis` default is `-1` (aligns with tf/torch default behavior).
- Passing `axis=None` explicitly mean a reduction on all axes, which fits with jax and all numpy reduction ops.

This should be least surprising overall to users.

Also, let's `reshape` to get the `None` behavior we want in torch/tf, rather then compute the reduction ourselves, so we can be sure our numerics are the same close to zero.

### CODE REVIEW

File: keras_core/backend/tensorflow/nn.py [hunk 1]
```diff
@@ -74,26 +74,27 @@ def gelu(x, approximate=True):
...
         # Unlike numpy, tf will handle axis=None as axis=-1.
         # We need this workaround for the reduction on every dim.
-        logits_exp = tf.exp(logits)
-        output = logits_exp / tf.reduce_sum(logits_exp, keepdims=True)
+        output = tf.reshape(x, [-1])
+        output = tf.nn.softmax(output, axis=-1)
+        output = tf.reshape(output, tf.shape(x))
     else:
         output = tf.nn.softmax(x, axis=axis)
     output._keras_logits = logits
```
Comment (mattdangerw) [inline keras_core/backend/tensorflow/nn.py:87]: why is this needed btw?

Comment (fchollet) [reply keras_core/backend/tensorflow/nn.py:87]: When computing crossentropy, we try to do it from logits rather from softmax/sigmoid values.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
